crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -839,40 +879,339 @@ fn test_simple_nested_fpi() {
+                # => [storage_value]
+
+                exec.sys::truncate_stack
+            end
+            ",
+            foreign_account_proc_hash = foreign_accounts.last().unwrap().code().procedures()[0].mast_root(),
+            foreign_prefix = foreign_accounts.last().unwrap().id().prefix().as_felt(),
+            foreign_suffix = foreign_accounts.last().unwrap().id().suffix(),
+        );
+
+        let tx_script = TransactionScript::compile(
+            code,
+            vec![],
+            TransactionKernel::testing_assembler().with_debug_mode(true),
+        )
+        .unwrap();
+
+        let tx_context = mock_chain
+            .build_tx_context(native_account.id(), &[], &[])
+            .advice_inputs(advice_inputs.clone())
+            .tx_script(tx_script)
+            .build();
+
+        let block_ref = tx_context.tx_inputs().block_header().block_num();
+        let note_ids = tx_context
+            .tx_inputs()
+            .input_notes()
+            .iter()
+            .map(|note| note.id())
+            .collect::<Vec<_>>();
+
+        let mut executor = TransactionExecutor::new(tx_context.get_data_store(), None)
+            .with_tracing()
+            .with_debug_mode();
+
+        // load the mast forest of the foreign account's code to be able to create an account 
+        // procedure index map and execute the specified foreign procedure
+        for foreign_account in foreign_accounts {
+            executor.load_account_code(foreign_account.code());
+        }
+
+        let err = executor
+            .execute_transaction(
+                native_account.id(),
+                block_ref,
+                &note_ids,
+                tx_context.tx_args().clone(),
+            ).unwrap_err();
+
+        let TransactionExecutorError::TransactionProgramExecutionFailed(err) = err else {
+            panic!("unexpected error")
+        };
+
+        assert_execution_error!(Err::<(), _>(err), ERR_FOREIGN_ACCOUNT_MAX_NUMBER_EXCEEDED);
+    }).expect("tread panic external").join().expect("tread panic internal");
+}
+
+/// Test that code will panic in attempt to call a procedure from the native account.
+#[test]
+fn test_nested_fpi_native_account_invocation() {
+    // ------ FIRST FOREIGN ACCOUNT ---------------------------------------------------------------
+    let foreign_account_code_source = "
+        use.miden::tx
+
+        use.std::sys
+
+        export.first_account_foreign_proc
+            # get the storage item at index 0
+            # pad the stack for the `execute_foreign_procedure` execution
+            padw padw padw push.0.0.0
+            # => [pad(15)]
+
+            # get the hash of the native account procedure from the advice stack
+            adv_push.4
+
+            # push the ID of the native account from the advice stack
+            adv_push.2
+            # => [native_account_id_prefix, native_account_id_suffix, NATIVE_PROC_ROOT, pad(15)]
```

Comment:
I guess we could also pass the procref of some mock account procedure as well as the account ID directly on the stack as arguments. Do you think that would improve readability of test?